### ISSUE TITLE
App Transport Security Issue

### ISSUE THREAD

**Issue by fdtorres1** [BoltAI/BoltAI#48]
I can't tell if this is a me issue or the most recent 1.15.1 update. I am now getting an error return on any submissions to GPT-4 with Google Search enabled. This was working prior to the most recent update but I'm not sure if this is something in the update or a change with Google, or...

**Comment by fdtorres1** [comment]
<img width="749" alt="Screenshot 2024-03-15 at 3 18 41 PM" src="https://github.com/BoltAI/BoltAI/assets/163592391/f60130f6-907e-42bd-850e-f48fa70622b9">

**Comment by longseespace** [comment]
Thanks for the bug report. This is due to the Apple's App Transport Security Policy. By default, Apple requires all connection must be via a secure connection (ie: HTTPS or SSL).

It's likely that the google search returned a non-secure webpage and it cause this issue.

I will take a deeper look, thanks.

**Comment by fdtorres1** [comment]
Ok, I've tested it with a few other queries and it looks like it's possibly a subject matter issue? My specific query is: "Is there a typical cost estimate for an indoor pyrotechnics display for about 10 minutes in length?" I assume it's an issue with the word pyrotechnics. It also didn't work with fireworks. I tested with a generic question about the average temperature in a location and the search worked fine for that. I can find the information elsewhere but wanted to share what I discovered. Not sure if it's an Apple thing or OpenAI at this point.
